app/controllers/pensions/transferIntoOverseasPensions/OverseasTransferChargePaidController.scala

```diff
@@ -41,35 +42,53 @@ class OverseasTransferChargePaidController @Inject()(actionsProvider: ActionsPro
+        pageView(OverseasTransferChargePaidPage(taxYear, pensionSchemeIndex, sessionUserData.pensionsUserData.pensions.transfersIntoOverseasPensions, formsProvider.overseasTransferChargePaidForm)))
+    }
   }
 
-  def submit(taxYear: Int): Action[AnyContent] = {
-    actionsProvider.userSessionDataFor(taxYear).async { implicit request =>
-      formsProvider.overseasTransferChargePaidForm.bindFromRequest().fold(
-        formWithErrors =>
-          Future.successful(
-            BadRequest(pageView(OverseasTransferChargePaidPage(taxYear, request.pensionsUserData.pensions.transfersIntoOverseasPensions, formWithErrors)))),
-        yesNoValue => {
-          overseasTransferChargesService.updateOverseasTransferChargeQuestion(request.pensionsUserData, yesNoValue).map {
-            case Left(_) => errorHandler.internalServerError()
-            case Right(userData) => Redirect(getRedirectCall(taxYear, yesNoValue, userData))
+  def submit(taxYear: Int, pensionSchemeIndex: Option[Int]): Action[AnyContent] = {
+    actionsProvider.userSessionDataFor(taxYear).async { implicit sessionUserData =>
+
+      validateScheme(pensionSchemeIndex, sessionUserData.pensionsUserData.pensions.transfersIntoOverseasPensions.transferPensionScheme) match {
+        case Left(_) => Future.successful(outOfBoundsRedirect(taxYear))
+        case Right(_) => formsProvider.overseasTransferChargePaidForm.bindFromRequest().fold(
+          formWithErrors =>
+            Future.successful(
+              BadRequest(pageView(OverseasTransferChargePaidPage(taxYear, pensionSchemeIndex, sessionUserData.pensionsUserData.pensions.transfersIntoOverseasPensions, formWithErrors)))),
+          yesNoValue => {
+            overseasTransferChargesService.updateOverseasTransferChargeQuestion(sessionUserData.pensionsUserData, yesNoValue, pensionSchemeIndex).map {
+              case Left(_) => errorHandler.internalServerError()
+              case Right(userData) => Redirect(getRedirectCall(taxYear, pensionSchemeIndex, yesNoValue, userData))
+            }
           }
-        }
-      )
+        )
+      }
     }
   }
 
+  def validateScheme(index: Option[Int], pensionSchemeLists: Seq[TransferPensionScheme]): Either[Unit, Option[TransferPensionScheme]] = {
+    index match {
+      case Some(value) if pensionSchemeLists.indices contains value => Right(Some(pensionSchemeLists(value)))
+      case None => Right(None)
+      case _ => Left(())
+    }
+  }
 
   private def getRedirectCall(taxYear: Int,
+                              pensionSchemeIndex: Option[Int],
                               yesNoValue: Boolean,
                               userData: PensionsUserData): Call = {
     if (yesNoValue) {
-      controllers.pensions.transferIntoOverseasPensions.routes.OverseasTransferChargePaidController.show(taxYear) //Redirect to
+      controllers.pensions.transferIntoOverseasPensions.routes.OverseasTransferChargePaidController.show(taxYear, pensionSchemeIndex) //Redirect to
```

Comment:
When creating adding a new Pension scheme pensionSchemeIndex = None but the size of pensionSchemeLists has increased. How will the next page know which pension scheme has been added with the UKCharge question in order to set the remaining pension scheme values